lute/main.py

```diff
@@ -108,12 +111,34 @@ def _start(args):
         "--config",
         help="Path to override config file.  Uses lute/config/config.yml if not set.",
     )
+    parser.add_argument(
+        "--upgrade",
+        action="store_true",
+        help="Upgrade to latest version (doesn't work with docker)",
+    )
+
     try:
-        _start(parser.parse_args())
+        args = parser.parse_args()
+        if args.upgrade:
```

Comment:
This command means that lute.main has different behaviours based on flags: `python -m lute.main --upgrade --port 5000` won't actually start lute, it only upgrades.  Should the `--upgrade` flag first run the update process, and then start lute?

The more I think about this, the more I feel that the "update" process should be handled outside of lute.main, and inside of any wrapper script that people use to start Lute.  For example, for my mac, I have the following:

```
#!/bin/bash

# Start venv, pull a version from pypi if specified, and start lute with this config.

source .venv/bin/activate

VERSION="$1"
if [[ -z "$VERSION" ]]; then
    echo "Using existing install."
else
    echo "Pulling version $VERSION"
    pip install --upgrade lute3==${VERSION}
fi

open http://localhost:9876
python -m lute.main --port 9876
```